sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2036,10 +2034,15 @@ void ProgramManager::removeImages(sycl_device_binaries DeviceBinary) {
         continue;
       }
 
-      auto It = m_KernelName2KernelIDs.find(EntriesIt->name);
-      assert(It != m_KernelName2KernelIDs.end());
-      m_KernelName2KernelIDs.erase(It);
-      m_KernelIDs2BinImage.erase(It->second);
+      // remove Everything associated with this KernelName
```

Comment:
```suggestion
      // remove everything associated with this KernelName
```

Reply:
done in https://github.com/intel/llvm/pull/17091/commits/42a642f27e8e935da0f19205cbe6c5bc1d1ff2b5